core/mempool/src/lib.rs

```diff
@@ -66,7 +69,17 @@ impl Pool {
     }
 
     pub fn is_empty(&self) -> bool {
-        self.transactions.read().expect(POISONED_LOCK_ERR).is_empty() && self.receipts.read().expect(POISONED_LOCK_ERR).is_empty()
+        self.transactions.read().expect(POISONED_LOCK_ERR).is_empty() 
+        && self.receipts.read().expect(POISONED_LOCK_ERR).is_empty()
+    }
+
+    pub fn len(&self) -> usize {
```

Comment:
Is this planned to be used for testing only? If yes then we should move it next to the tests, below.

Reply:
No. While it is only used in tests for now, I think it makes sense for expose `len` as part of the API.